src/haddock/engine.py

```diff
@@ -1,10 +1,38 @@
 """Running CNS scripts"""
 import subprocess
-from haddock.error import CNSRunningError
+from haddock.error import CNSRunningError, JobRunningError
 from haddock.parallel import Scheduler
+import shlex
 from haddock.defaults import CNS_EXE, NUM_CORES
 
 
+class Job:
+    """A job to be executed by the engine"""
+    def __init__(self, input, output, executable, *args):
+        self.input = input
+        self.output = output
+        self.executable = executable
+        self.args = args
+
+    def run(self):
+        if self.args:
+            cmd = f"{self.executable} {' '.join(map(str, self.args))}"
+            self.executable = shlex.split(cmd)
+
+        # capture output to stdout
+        cmd += f' {self.input}'
```

Comment:
I think here there's a bug if `args` are not given because `cmd` won't be defined.

Reply:
oh yeh, indeed!